assets/src/edit-story/components/fontPicker/fontPicker.js

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useState, useCallback, useRef } from 'react';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+import { rgba } from 'polished';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { useDebouncedCallback } from 'use-debounce';
+import { ReactComponent as DropDownIcon } from '../../icons/dropdown.svg';
+import Popup from '../popup';
+import FontPickerContainer from './pickerContainer';
+
+const DEFAULT_WIDTH = 240;
+
+const Container = styled.div`
+  display: flex;
+  flex-direction: column;
+  flex-grow: 1;
+  color: ${({ theme }) => theme.colors.fg.v0};
+  font-family: ${({ theme }) => theme.fonts.body1.font};
+`;
+
+const FontPickerSelect = styled.div.attrs({ role: 'button', tabIndex: '0' })`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: center;
+  flex-grow: 1;
+  background-color: ${({ theme, lightMode }) =>
+    lightMode ? rgba(theme.colors.fg.v1, 0.1) : rgba(theme.colors.bg.v0, 0.3)};
+  border-radius: 4px;
+  padding: 2px 0 2px 6px;
+  cursor: pointer;
+
+  ${({ disabled }) =>
+    disabled &&
+    `
+      pointer-events: none;
+      opacity: 0.3;
+    `}
+
+  svg {
+    width: 28px;
+    height: 28px;
+    color: ${({ theme, lightMode }) =>
+      lightMode ? theme.colors.fg.v1 : rgba(theme.colors.fg.v1, 0.3)};
+  }
+`;
+
+const FontPickerTitle = styled.span`
+  user-select: none;
+  color: ${({ theme }) => theme.colors.fg.v1};
+  font-family: ${({ theme }) => theme.fonts.label.family};
+  font-size: ${({ theme }) => theme.fonts.label.size};
+  line-height: ${({ theme }) => theme.fonts.label.lineHeight};
+  font-weight: ${({ theme }) => theme.fonts.label.weight};
+  letter-spacing: ${({ theme }) => theme.fonts.label.letterSpacing};
+`;
+
+function FontPicker({ onChange, lightMode = false, placeholder, value }) {
+  const selectRef = useRef();
+
+  const [isOpen, setIsOpen] = useState(false);
+
+  // Make some delay, since when click the dropdown again to close it, this function called before by useFocusOut
+  // Causing reopen
+  const [closeFontPicker] = useDebouncedCallback(() => {
```

Comment:
Why does the close callback need to be debounced?

Reply:
hmm.. Yes this should not be a debounced. But maybe setTimeout will do the work. I think it's kind of similar issue Morten posted on amp stories. We have select button and the main dropdown with fonts is a pop up menu. When the user click the select button, and the picker will show up. And if the user click it again, it should be closed. But the main font picker container with dropdown has useFocusOut event which will be fired before the button select. So when the select button click event happens, the font picker already closed, so it opens the font picker again. To make useFoucsOut event fired after select button click event, I added minor delay for it.

Reply:
I think `setTimeout` is ok.